CMakeLists.txt

```diff
@@ -6,6 +6,9 @@ project(wav2letter++)
 set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_LINKER_WRAPPER_FLAG "-Xlinker" " ")
```

Comment:
@tbfly — it's not clear to me why these are needed: can you explain? Which version of CMake/gcc/ld,gold are you using that this is required, and what's the behavior without them? These are also included [by default](https://github.com/Kitware/CMake/blob/master/Modules/Compiler/HP-CXX.cmake#L6) in most cases.

You're also overwriting the value of CMAKE_CXX_LINKER_WRAPPER_FLAG immediately below...

Reply:
@jacobkahn 
gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1~18.04) 
cmake version 3.10.2

Reply:
It seems it can just take 
`set(CMAKE_CXX_LINKER_WRAPPER_FLAG "-Xlinker" " ")`
or
`set(CMAKE_CXX_LINKER_WRAPPER_FLAG "-Wl,")
set(CMAKE_CXX_LINKER_WRAPPER_FLAG_SEP ",")
`

Reply:
If I don't add it in Ubuntu 18.04.2, it will cause Decoder Link error as following:

> CMakeFiles/Train.dir/src/lm/KenLM.cpp.o: In function w2l::KenLM::KenLM(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)': KenLM.cpp:(.text+0xfa): undefined reference to lm::ngram::Config::Config()'
KenLM.cpp:(.text+0x123): undefined reference to `lm::ngram::LoadVirtual(char const*, lm::ngram::Config const&, lm::ngram::ModelType)'
collect2: error: ld returned 1 exit status
CMakeFiles/Train.dir/build.make:1681: recipe for target 'Train' failed
make[2]: *** [Train] Error 1
CMakeFiles/Makefile2:70: recipe for target 'CMakeFiles/Train.dir/all' failed
make[1]: *** [CMakeFiles/Train.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

Reply:
I just checkout clean code, and cmake and compile it again. Now, I can't repeat the link error again. So I think it is just my build system envirment issue. You may omit it.